src/hotspot/share/runtime/suspendResumeManager.cpp

```diff
@@ -81,11 +81,12 @@ void SuspendResumeManager::set_suspended(bool is_suspend, bool register_vthread_
   AtomicAccess::store(&_suspended, is_suspend);
 }
 
-void SuspendResumeManager::set_suspended_with_id(int64_t id, bool register_vthread_SR) {
+void SuspendResumeManager::set_suspended_current_thread(int64_t vthread_id, bool register_vthread_SR) {
```

Comment:
It was totally not at all apparent that `_target` had to be the current thread here! Is it always the current thread? Won't this variant only ever get called when `register_vthread_SR` would be true - in which case we don't need that parameter?

Reply:
Yes, the id is always extracted and the exposure of "raw" id on this level is not the good desing. However, I don't know how to implement this better.